pe-parser-library/src/parse.cpp

```diff
@@ -1881,20 +1881,23 @@ bool getDebugDir(parsed_pe *p) {
         rawData =
             curEnt.AddressOfRawData + p->peHeader.nt.OptionalHeader64.ImageBase;
       } else {
-        return false;
+        break;
       }
 
       //
       // Get the section for the data
       //
       section dataSec;
       if (!getSecForVA(p->internal->secs, rawData, dataSec)) {
-        return false;
+        break;
       }
 
       debugent ent;
 
       auto dataofft = static_cast<std::uint32_t>(rawData - dataSec.sectionBase);
+      if (dataofft + curEnt.SizeOfData > dataSec.sectionData->bufLen) {
+        break;
+      }
```

Comment:
Minor nitpick: I'd be good to have comments on these `break`s explaining that we silently ignore invalid Debug Directory entries, rather than failing on them.

Reply:
I've added comments and update the code in the review.